SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialActorChannel.h

```diff
@@ -342,4 +344,11 @@ class SPATIALGDK_API USpatialActorChannel : public UActorChannel
 	// before the actor holding the position for all the hierarchy, it can immediately attempt to migrate back.
 	// Using this timestamp, we can back off attempting migrations for a while.
 	uint64 AuthorityReceivedTimestamp;
+
+	// Contains the frame at which the actor (or its owner) was seen outside its worker's authoritative region.
+	// Used to delay migration by one frame, to allow Pre-Replication events to take place in the regular replication loop.
+	uint32 MigrationFrame;
+
+	// WorkerId to migrate to.
+	VirtualWorkerId MigrationVirtualWorkerId = SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
```

Comment:
This feels a little bit weird. We make the decision on the target migration worker on the frame before, but ignore any changes to the Actor that could affect migration on the next frame. I think forcing net update on the whole hierarchy is necessary, but we may need to evaluate the migration virtual worker ID on the actual migration frame.

Reply:
Is one frame going to make such a difference ? If it would, it means we have some jittering between two zones, and I think we should avoid that by having hysterisis transistions.

Reply:
Actually we should at least take into account if the actor is locked.

Reply:
I agree that if 1 frame makes a difference, there's other problems, but I feel like this change adds additional potential surprise to migration behaviour. What's the reason for determining the ID in advance? Seems to me like evaluating it when migration actually happens wouldn't be very complicated, but I might be missing something.

Reply:
We have hysteresis already form the migration backoff time. I still believe we should at least reevaluate the hierarchy authority, even if it's one frame. Currently, it's entirely possible that WhoShouldHaveAuthority returns worker X but on the same frame we set authority intent to worker Y, I don't think this okay to leave open.

Reply:
Waiting a tick isn't that different from just not having synchronous load balancing.
Right now I agree you should determine the worker at the last possible moment.
Broadly speaking if we are worried about actors being changed such that the worker they are moving towards can not simulate them at all then we should have workers inspect certain parts of entity state before creating the actor to determine if it can actually simulate it (this can potentially be done using completeness).
The odds are you want to do this anyway if you can have things move into a dungeon phase. Worth noting this also places hard limits on what a user can or should be able to do with locking.

TL;DR this doesn't really introduce new problems but I agree that we should work out what worker is being moved to when it is moved.

Reply:
Changed to get the destination worker right before migration.